src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImpl.cs

```diff
@@ -8835,35 +9057,40 @@ private void ThrowExpectingWhitespace(int pos)
 
         private int GetIndexOfAttributeWithoutPrefix(string name)
         {
-            name = _nameTable.Get(name);
-            if (name == null)
+            string? atomizedName = _nameTable.Get(name);
+            if (atomizedName == null)
             {
                 return -1;
             }
+
             for (int i = _index + 1; i < _index + _attrCount + 1; i++)
             {
-                if (Ref.Equal(_nodes[i].localName, name) && _nodes[i].prefix.Length == 0)
+                if (Ref.Equal(_nodes[i].localName, atomizedName) && _nodes[i].prefix.Length == 0)
                 {
                     return i;
                 }
             }
+
             return -1;
         }
 
         private int GetIndexOfAttributeWithPrefix(string name)
         {
             name = _nameTable.Add(name);
```

Comment:
I don't want to change it as part of this PR. I got https://github.com/dotnet/runtime/issues/35213 to track that. I'm not 100% sure about this since this code is really fragile. It will be easier to find it if it's separate if we somehow regress this